pkg/db-repo/repository.go

```diff
@@ -126,6 +134,14 @@ func (r *repository) Update(ctx context.Context, value ModelBased) error {
 
 	err := r.orm.Save(value).Error
 
+	isDuplicate := db.IsDuplicateEntryError(err)
+	if isDuplicate {
+		logger.Warnf("could not update model of type %s with id %d due to duplicate entry error: %s", modelId, *value.GetId(), err.Error())
+		return &db.DuplicateEntryError{
+			Err: err,
+		}
+	}
+
 	if err != nil {
 		logger.Errorf(err, "could not update model of type %s with id %d", modelId, *value.GetId())
```

Comment:
```suggestion
		logger.Errorf(err, "could not update model of type %s with id %d", modelId, mdl.EmptyUintIfNil(value.GetId()))
```

Reply:
maybe also check other pointer derefs for safeness. I know that there shouldn't be a model passed to Update that doesn't have a Id, but you could

Reply:
I think you should not get past this error check should your id be nil. Either it is nil and then it is treated like a create by `Save` or it is not nil and then the problem does not exist. In case of a create, we either succeed and get an id set, or we fail and have one of the two error cases. So I agree, we should guard against nil at the two places you mentioned, but there shouldn't be any other possible places.